packages/flutter/test/material/flexible_space_bar_test.dart

```diff
@@ -470,6 +470,57 @@ void main() {
     );
   });
 
+  testWidgets('FlexibleSpaceBar sets width constraints for the title - override expandedScale', (WidgetTester tester) async {
+    const double titleFontSize = 20.0;
+    const double height = 300.0;
+    late double width;
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Scaffold(
+          body: Builder(
+            builder: (BuildContext context) {
+              width = MediaQuery.of(context).size.width;
+              return CustomScrollView(
+                slivers: <Widget>[
+                  SliverAppBar(
+                    expandedHeight: height,
+                    pinned: true,
+                    stretch: true,
+                    flexibleSpace: FlexibleSpaceBar(
+                      expandedScale: 3,
+                      titlePadding: EdgeInsets.zero,
+                      title: Text(
+                        'X' * 2000,
```

Comment:
Making the test layout 2000 'X' characters seems unnecessary. 

Don't we want to verify that the title grows in X and Y?

Reply:
This test makes sure the **constraints** are set properly so the text will not overflow on `actions` for example. As far as I could tell there really is no straightforward way to check that the title has grown in some physical size after the transform.

It doesn't really matter when you check the rect (or render object constraints for that matter) if its something like a single character `X`, it will just return a rect with width and height equal to the font size. The only way to tell them apart is that the constraint for the max-width will be divided by the tweened `expandedTitleScale`, and the top and bottom of the rect will be offset by 
`+ font size * (1 - tweened expandedTitleScale)`

This is why the text is 2000 chars long (like in the test I copied this from) since its just to make sure it overflows to test the width constraint. For reference yes I'm pretty sure it can be 41 chars instead to overflow the collapsed full width of 800.

Reply:
I see. Sorry I'd forgotten that the effect of the scale transform only impacts what's painted.  I suppose you could check what's being painted with the mock canvas stuff (`expect(find ..., paints.. ...)`), but a golden image test would probably be appropriate and a little easier to understand.